Doppler.Sap/Factory/CreateOrUpdateBusinessPartnerHandler.cs

```diff
@@ -29,14 +29,26 @@ public class CreateOrUpdateBusinessPartnerHandler
 
         public async Task<SapTaskResult> Handle(SapTask dequeuedTask)
         {
-            var sapSystem = SapSystemHelper.GetSapSystemByBillingSystem(dequeuedTask.DopplerUser.BillingSystemId);
+            try
+            {
+                var sapSystem = SapSystemHelper.GetSapSystemByBillingSystem(dequeuedTask.DopplerUser.BillingSystemId);
 
-            var sapTaskHandler = _sapServiceSettingsFactory.CreateHandler(sapSystem);
-            dequeuedTask = await sapTaskHandler.CreateBusinessPartnerFromDopplerUser(dequeuedTask);
+                var sapTaskHandler = _sapServiceSettingsFactory.CreateHandler(sapSystem);
+                dequeuedTask = await sapTaskHandler.CreateBusinessPartnerFromDopplerUser(dequeuedTask);
 
-            return !dequeuedTask.ExistentBusinessPartner.CreateDate.HasValue ?
-                await CreateBusinessPartner(dequeuedTask, sapSystem) :
-                await UpdateBusinessPartner(dequeuedTask, sapSystem);
+                return !dequeuedTask.ExistentBusinessPartner.CreateDate.HasValue ?
+                    await CreateBusinessPartner(dequeuedTask, sapSystem) :
+                    await UpdateBusinessPartner(dequeuedTask, sapSystem);
+            }
+            catch (Exception ex)
+            {
+                return new SapTaskResult
+                {
+                    IsSuccessful = false,
+                    SapResponseContent = ex.Message,
```

Comment:
What do you think to add the user that have the error?. For example: 
```$"Failed to create or update the Business Partner for the user: {dequeuedTask.BusinessPartner.CardCode}. Error: {ex.Message}"```

Reply:
Implement the message as you suggested, I had to modify a test to set de BusinessPartner.CardCode in the mock